### ISSUE TITLE
Display for `SequenceOptions` in `AlterColumn` shouldn't check whether the option list is empty, Min and Max shouldn't be listed as `Empty` in Sequence Opt List

### ISSUE THREAD

**Issue by PartiallyTyped** [sqlparser-rs/sqlparser-rs#1105]
Here, the options are defined as `AS IDENTITY [ ( sequence_options ) ]`, which is to say, if parentheses exist, then there's an option list, regardless of whether it is empty.

https://github.com/sqlparser-rs/sqlparser-rs/blob/b0b62887a2bbc464642a17e590b259411bf1c11d/src/ast/ddl.rs#L318-L323

This is correctly implemented here:

https://github.com/sqlparser-rs/sqlparser-rs/blob/b0b62887a2bbc464642a17e590b259411bf1c11d/src/parser/mod.rs#L4949-L4953

Thus, when `sequence_options` is `Some(_)` in

```rust
AlterColumnOperation::AddGenerated {
    generated_as,
    sequence_options
}
```
we expect `Display` to print at least the brackets regardless of the contents. 

However, the implementation for display also checks whether the option list is empty.

https://github.com/sqlparser-rs/sqlparser-rs/blob/b0b62887a2bbc464642a17e590b259411bf1c11d/src/ast/ddl.rs#L355-L368

This plays with an issue in `SequenceOption` parsing, where the `Min` and `Max` fields are given as `empty` regardless of whether they are there:

https://github.com/sqlparser-rs/sqlparser-rs/blob/b0b62887a2bbc464642a17e590b259411bf1c11d/src/parser/mod.rs#L8565-L8583

The tag `empty` is used to remove them in display.

In conclusion, if parentheses exist, then `Min` and `Max` options always make it to the options list therefore Display passes checks and parentheses are printed, then the two fields are not printed because the Display for an empty tag is an empty string. In the case where the parentheses don't exist, the check correctly skips that.

In my opinion, the two fields shouldn't be included in the opt list, and the opt list shouldn't be checked for emptiness.

I will open a PR for this.
